File: typescript/src/game-finder.ts
import fetch from "node-fetch";
import { config } from "./config";

export class GameFinder {
    // prettier-ignore
    private static readonly LOAD_BALANCER_URL = `http://${config.loadBalancerHost}:${config.loadBalancerPort}`;

    static async next(): Promise<string> {
        return fetch(this.LOAD_BALANCER_URL + "/join-game").then(response =>
            response.text()
        );
    }
}

File: python3/src/gameFinder.py
import urllib.request
from config import LOAD_BALANCER_HOST, LOAD_BALANCER_PORT

class GameFinder:
    LOAD_BALANCER_URL = "http://" + LOAD_BALANCER_HOST + ":" + LOAD_BALANCER_PORT

    @staticmethod
    def next():
        return urllib.request.urlopen(GameFinder.LOAD_BALANCER_URL + "/join-game").read()
